### ISSUE TITLE
Distribute the corporate income tax

### ISSUE THREAD

**Issue by MattHJensen** [OpenSourcePolicyCenter/Tax-Calculator#223]
We do not currently distribute the corporate income tax in our microsim model. We should. 

Useful docs

- [OTA](http://www.treasury.gov/resource-center/tax-policy/tax-analysis/Documents/OTA-T2012-05-Distributing-the-Corporate-Income-Tax-Methodology-May-2012.pdf)
- [TPC](http://www.taxpolicycenter.org/UploadedPDF/412651-Tax-Model-Corporate-Tax-Incidence.pdf)
- [JCT](https://www.jct.gov/publications.html?func=showdown&id=4528)
- [CBO](http://www.cbo.gov/sites/default/files/cbofiles/attachments/43373-AverageTaxRates_screen.pdf)

**Comment by mmessick** [comment]
@iterate_jit(parameters=['revenue_collected', 'percent_labor',
                             'percent_supernormal', 'percent_normal'],
                 nopython=True, puf=True)
    def Dist_Corp_Inc_Tax(revenue_collected, percent_labor, percent_supernormal,
                      percent_normal, agg_comp, agg_dividends, agg_capgains,
                      agg_bonds, agg_self_employed, corp_tax_burden,
                      dividends, netcapgains, bonds, compensation, e09400):

        # self-employment (e09400)
        # individual's share of total self-employment
        self_employed = e09400 / agg_self_employed

        share_of_divs = dividends / agg_dividends
        share_of_capgains = netcapgains / agg_capgains

        # adapted from the JCT
        share_from_labor = (percent_labor * revenue_collected *
                                   compensation / agg_comp)

        # adapted from the JCT and TPC
        share_from_normal = ((percent_normal * revenue_collected) *
                                        (share_of_divs * .4 + share_of_capgains * .4 +
                                        self_employed * .4 + bonds / agg_bonds))

        share_from_supernormal = (percent_supernormal * revenue_collected
                                       * (share_of_divs * .6 + share_of_capgains * .6))

        # individual's share of the corporate income tax
        share_corptax_burden = share_from_labor + share_from_normal + share_from_supernormal

        # incorporating the burden into our measure of expanded income
        # expanded_income = expanded_income + corp_tax_burden

        return share_corptax_burden


This function calculates the burden of the corporate income tax borne
by an individual in the economy. It also incorporates this burden into
our measure of the individual's expanded income.

Parameters
----------------
    revenue_collected : float
        value of yearly revenue estimated to be collected in aggregate from the
        corporate income tax assigned by the user
    percent_labor : float
        percent of the burden that will be borne by labor
    percent_normal : float
        percent of the burden that will be borne by normal returns
    percent_supernormal : float
        percent of the burden that will be borne by supernormal returns

Returns
-----------
    corp_tax_burden : float
        individual's share of the corporate income tax
    expanded_income : float
        the updated expanded income

**Comment by mmessick** [comment]
So for the code above, I am trying to call it using:
    Dist_Corp_Inc_Tax(agg_comp, agg_dividends, agg_capgains,
                      agg_bonds, agg_self_employed, calc.params, calc.records)

Where agg_comp, etc are values I am giving it and I have created a working calc object.
However, I keep getting this error and I'm not sure what is going wrong:
    raise ValueError("Unknown arg: " + farg)
    ValueError: Unknown arg: share_corptax_burden

I added share_corptax_burden to the zeroed_names array in records.py, which I thought would fix the problem, but apparently not. @talumbau you said you might be able to help clarify part of the @iterator_jit decorator to me, so maybe I am just using it wrong?
